.github/workflows/check.yml

```diff
@@ -74,6 +74,62 @@ jobs:
           flags: tests
           name: ${{ matrix.py }} - ${{ matrix.os }}
 
+  test-android:
+    name: test ${{ matrix.py }} - Android
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        py:
+          - "3.10.0-alpha.7"
+          - "3.9"
+          - "3.8"
+          - "3.7"
+          - "3.6"
+          - "pypy-3.6"
+          - "pypy-3.7"
+    steps:
+      - name: Setup python for tox
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.9"
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Install tox
+        run: python -m pip install tox
+      - name: Setup python for test ${{ matrix.py }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.py }}
+      - name: Pick environment to run
+        run: |
+          import json
+          import os
+          import subprocess
+          major, minor, impl = json.loads(subprocess.check_output(["python", "-c", "import json; import sys; import platform; print(json.dumps([sys.version_info[0], sys.version_info[1], platform.python_implementation()]));"], universal_newlines=True))
+          with open(os.environ['GITHUB_ENV'], 'a') as file_handler:
+                file_handler.write('TOXENV=' + ("py" if impl == "CPython" else "pypy") + ("{}{}".format(major, minor) if impl == "CPython" else "3") + "\n")
+        shell: python
+      - name: Setup test suite
+        run: tox -vv --notest
+      - name: Run test suite
+        run: tox --skip-pkg-install
+        env:
+          PYTEST_ADDOPTS: "-vv --durations=20"
+          ANDROID_DATA: /data
```

Comment:
You can use monkeypatch.setenv fixture to set these for android tests, no need to add new targets.